cli/src/cmd_auth.rs

```diff
@@ -267,84 +291,104 @@ impl CmdAuthLogin {
+            .entry("profile")
+            .or_insert_with(|| {
+                let mut table = Table::default();
+                // Avoid a bare [profile] table.
+                table.set_implicit(true);
+                Item::Table(table)
+            })
+            .as_table_mut()
+            .unwrap_or_else(|| {
+                panic!(
+                    "\"profile\" in {} is not a table",
+                    credentials_path.to_string_lossy()
+                )
+            });
+
+        let profile = profile_table
+            .entry(profile_name)
+            .or_insert_with(|| Item::Table(Table::default()))
+            .as_table_mut()
+            .unwrap_or_else(|| {
+                panic!(
+                    "\"profile.{}\" in {} is not a table",
+                    profile_name,
+                    credentials_path.to_string_lossy()
+                )
+            });
+
+        profile.insert("host", toml_edit::value(self.host.as_host().to_string()));
+        profile.insert("token", toml_edit::value(token));
+        profile.insert("user", toml_edit::value(uid.to_string()));
+
+        std::fs::write(credentials_path, credentials.to_string())
+            .expect("unable to write credentials.toml");
+
+        println!("{}", credentials);
+
+        // If there is no default profile then we'll set this new profile to be
+        // the default.
+        if ctx.config_file().basics.default_profile.is_none() {
+            let config_path = ctx.client_config().config_dir().join("config.toml");
+            let mut config = if let Ok(contents) = std::fs::read_to_string(config_path.clone()) {
+                contents.parse::<toml_edit::DocumentMut>().unwrap()
+            } else {
+                Default::default()
+            };
 
-        let host_entry = Host {
-            token,
-            user: uid.to_string(),
-            default: false,
-        };
+            config.insert("default-profile", Item::Value(profile_name.into()));
 
-        ctx.config()
-            .update_host(self.host.to_string(), host_entry)?;
+            std::fs::write(config_path, config.to_string()).expect("unable to write config.toml");
+        }
 
         Ok(())
     }
 }
 
-/// Removes saved authentication information.
+/// Removes saved authentication information from profiles.
 ///
-/// This command does not invalidate any tokens from the hosts.
+/// This command does not invalidate any tokens.
 #[derive(Parser, Debug, Clone)]
 #[command(verbatim_doc_comment)]
 pub struct CmdAuthLogout {
-    /// Remove authentication information for a single host.
-    #[clap(
-        short = 'H',
-        long,
-        value_parser = parse_host,
-        required_unless_present = "all",
-        conflicts_with = "all",
-    )]
-    pub host: Option<url::Url>,
-    /// If set, all known hosts and authentication information will be deleted.
+    /// If set, authentication information from all profiles will be deleted.
```

Comment:
Why is the ability to remove a single profile being removed?

Reply:
it's not.